File: python/samples/file_management_demo.py
import asyncio
from trinsic.proto.services.filemanagement.v1 import (
    DeleteFileRequest,
    GetFileRequest,
    GetStorageStatsRequest,
    ListFilesRequest,
    UploadFileRequest,
)

from trinsic.trinsic_service import TrinsicService
from trinsic.trinsic_util import trinsic_config, set_eventloop_policy


async def file_management_demo():
    config = trinsic_config()
    trinsic = TrinsicService(server_config=config)
    await trinsic.account.login_anonymous(ecosystem_id="default")

    # uploadFile() {
    # Get raw bytes of string
    file_bytes = bytes("Hello, world!", "utf-8")
    file_mime_type = "application/text"

    upload_response = await trinsic.file_management.upload_file(
        request=UploadFileRequest(file_bytes, file_mime_type)
    )
    # }

    file_id = upload_response.uploaded_file.id

    # getFile() {
    get_response = await trinsic.file_management.get_file(
        request=GetFileRequest(file_id)
    )
    # }

    # listFiles() {
    list_response = await trinsic.file_management.list_files(
        request=ListFilesRequest(
            query="SELECT * FROM _ ORDER BY _.uploadDate DESC OFFSET 0 LIMIT 100"
        )
    )
    # }

    assert list_response.files[0].id == file_id

    # getStorageStats() {
    get_stats_response = await trinsic.file_management.get_storage_stats(
        request=GetStorageStatsRequest()
    )
    # }

    # deleteFile() {
    await trinsic.file_management.delete_file(request=DeleteFileRequest(file_id))
    # }

    trinsic.close()


if __name__ == "__main__":
    set_eventloop_policy()
    asyncio.run(file_management_demo())

File: java/src/test/java/trinsic/FileManagementDemo.java
package trinsic;

import com.google.protobuf.ByteString;
import java.io.IOException;
import java.nio.charset.StandardCharsets;
import java.util.concurrent.ExecutionException;
import org.junit.jupiter.api.Assertions;
import trinsic.okapi.DidException;
import trinsic.services.TrinsicService;
import trinsic.services.filemanagement.v1.DeleteFileRequest;
import trinsic.services.filemanagement.v1.GetFileRequest;
import trinsic.services.filemanagement.v1.ListFilesRequest;
import trinsic.services.filemanagement.v1.UploadFileRequest;

public class FileManagementDemo {

  public static void main(String[] args)
      throws IOException, DidException, ExecutionException, InterruptedException {
    run();
  }

  public static void run()
      throws IOException, DidException, ExecutionException, InterruptedException {
    var trinsic = new TrinsicService(TrinsicUtilities.getTrinsicServiceOptions());
    var account = trinsic.account().loginAnonymous("default").get();

    trinsic.setAuthToken(account);

    // uploadFile() {
    // Get raw bytes of string
    byte[] fileBytes = "Hello, world!".getBytes(StandardCharsets.UTF_8);
    String fileMimeType = "application/text";

    var uploadResponse =
        trinsic
            .fileManagement()
            .uploadFile(
                UploadFileRequest.newBuilder()
                    .setContents(ByteString.copyFrom(fileBytes))
                    .setMimeType(fileMimeType)
                    .build())
            .get();
    // }

    var fileId = uploadResponse.getUploadedFile().getId();

    // getFile() {
    var getFileResponse =
        trinsic.fileManagement().getFile(GetFileRequest.newBuilder().setId(fileId).build()).get();
    // }

    // listFiles() {
    var listFilesResponse =
        trinsic
            .fileManagement()
            .listFiles(
                ListFilesRequest.newBuilder()
                    .setQuery("SELECT * FROM _ ORDER BY _.uploadDate DESC OFFSET 0 LIMIT 100")
                    .build())
            .get();
    // }

    Assertions.assertEquals(
        listFilesResponse.getFiles(0).getId(), uploadResponse.getUploadedFile().getId());

    // getStorageStats() {
    var getStatsResponse = trinsic.fileManagement().getStorageStats().get();
    // }

    Assertions.assertEquals(getStatsResponse.getStats().getNumFiles(), 1);
    Assertions.assertEquals(
        getStatsResponse.getStats().getTotalSize(), uploadResponse.getUploadedFile().getSize());

    // deleteFile() {
    trinsic.fileManagement().deleteFile(DeleteFileRequest.newBuilder().setId(fileId).build()).get();
    // }

    System.out.println("File Management demo successful");
    trinsic.shutdown();
  }
}
